### PR TITLE
HPCC-14578 Fix jlib warnings

### PR DESCRIPTION
**Pull request by richardkchapman**
Another warning that comes up on some builds.

Signed-off-by: Richard Chapman rchapman@hpccsystems.com

### CODE REVIEW

**Comment by mckellyln** [comment]
Looks good.

File: system/jlib/jdebug.cpp [hunk 1]
```diff
@@ -2246,14 +2246,19 @@ static class CMemoryUsageReporter: public Thread
...
         if (procfp) {
-            fscanf(procfp, "%lf %lf\n", &dbSystemTime, &dbIdleTime);
+            matched = fscanf(procfp, "%lf %lf\n", &dbSystemTime, &dbIdleTime);
             fclose(procfp);
             outofhandles = false;
         }
         latestCPU = unsigned(100.0 - (dbIdleTime - OldIdleTime)*100.0/(dbSystemTime - OldSystemTime) + 0.5);
-        OldSystemTime = dbSystemTime;
-        OldIdleTime = dbIdleTime;
+        if (procfp && matched == 2)
```
Comment (jakesmith) [inline system/jlib/jdebug.cpp]: the constructor does differently:
```
        if (!procfp || matched == 0 || matched == EOF)
            OldSystemTime = 0;
```

I think your change is better, in that in theory scanf could return 1 and the code should probably set to 0 if != 2 as you have here.
Perhaps both version should be:
```
if (!propfp || matched != 2)
{
    OldSystemTime = 0;
    OldIdleTime = 0;
}
```

Comment (richardkchapman) [inline system/jlib/jdebug.cpp]: Repushed to fix the other case

**Comment by jakesmith** [comment]
Looks good to merge
@richardkchapman

**Pull request merged**